newrelic-agent/src/main/java/com/newrelic/api/agent/NewRelicApiImplementation.java

```diff
@@ -574,6 +579,20 @@ public void setInstanceName(String instanceName) {
         MetricNames.recordApiSupportabilityMetric(MetricNames.SUPPORTABILITY_API_SET_INSTANCE_NAME);
     }
 
+    private static boolean isExpectedErrorConfigured(Throwable t) {
```

Comment:
The reason I'm doing this check here is because it is too late to check the config from anywhere in `ErrorServiceImpl`.

The agent spec requires that the `noticeError(String, boolean expected)` and similar APIs take precedence over config (where as the methods without the boolean need to check config).  If a user invokes the API and sets the boolean as false and it is configured to be expected via config, this should still report the error.

So, if we want to adhere to the spec for the `noticeError(String msg)` case, then we have to check config before calling the `noticeError(String msg, boolean expected)` method. Otherwise, there is no way to distinguish where the boolean was set.  OR we would need to do some significant refactoring and send the method signatures without the expected boolean down a different path.  I don't think it's really worth it.




<img width="1023" alt="expected-errors-api" src="https://user-images.githubusercontent.com/7401258/191409027-8d083132-874e-4116-96b6-7c2b4235a174.png">